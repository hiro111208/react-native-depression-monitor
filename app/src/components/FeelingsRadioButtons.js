import React, { useEffect, useState } from "react";
import { View, StyleSheet, TouchableOpacity, Image } from "react-native";
import verySad from "../../assets/images/verySad.png";
import sad from "../../assets/images/sad.png";
import neutral from "../../assets/images/neutral.png";
import happy from "../../assets/images/happy.png";
import veryHappy from "../../assets/images/veryHappy.png";
import colors from "../../config/colors";

//These are the buttons displayed on the LogFeelingScreen
const FeelingsRadioButtons = (props) => {
    var feelings = ["Very Happy", "Happy", "Neutral", "Sad", "Very Sad"];

    return (
        <View style={styles.radio}>
            {/* Render emojis for each feeling as buttons
          If selected the button will appear with a ring around it
         */}
            <TouchableOpacity
                style={
                    props.overallFeeling == "Very Sad" ? styles.selected : ""
                }
                onPress={() => {
                    props.setOverallFeeling("Very Sad");
                }}
                testID={"VERY_SAD"}
            >
                <Image style={styles.logo} source={verySad} />
            </TouchableOpacity>

            <TouchableOpacity
                style={props.overallFeeling == "Sad" ? styles.selected : ""}
                onPress={() => {
                    props.setOverallFeeling("Sad");
                }}
                testID={"SAD"}
            >
                <Image style={styles.logo} source={sad} />
            </TouchableOpacity>

            <TouchableOpacity
                style={props.overallFeeling == "Neutral" ? styles.selected : ""}
                onPress={() => {
                    props.setOverallFeeling("Neutral");
                }}
                testID={"NEUTRAL"}
            >
                <Image style={styles.logo} source={neutral} />
            </TouchableOpacity>

            <TouchableOpacity
                style={props.overallFeeling == "Happy" ? styles.selected : ""}
                onPress={() => {
                    props.setOverallFeeling("Happy");
                }}
                testID={"HAPPY"}
            >
                <Image style={styles.logo} source={happy} />
            </TouchableOpacity>

            <TouchableOpacity
                style={
                    props.overallFeeling == "Very Happy" ? styles.selected : ""
                }
                onPress={() => {
                    props.setOverallFeeling("Very Happy");
                }}
                testID={"VERY_HAPPY"}
            >
                <Image style={styles.logo} source={veryHappy} />
            </TouchableOpacity>
        </View>
    );
};

const styles = StyleSheet.create({
    radio: {
        padding: 10,
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "space-between",
        backgroundColor: "white",
        borderRadius: 50,
        borderColor: colors.darkBorder,
        borderWidth: 2,
        width: "95%",
    },
    logo: {
        width: 50,
        height: 50,
        marginHorizontal: 7,
        marginVertical: 7,
    },
    selected: {
        borderWidth: 2,
        borderColor: "rgba(0,0,0,0.2)",
        borderRadius: 100,
    },
});

export default FeelingsRadioButtons;
