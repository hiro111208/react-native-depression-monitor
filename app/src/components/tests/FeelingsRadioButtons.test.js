import React from "react";
import { shallow } from "enzyme";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import FeelingsRadioButtons from "../FeelingsRadioButtons";

describe("Testing FeelingsRadioButtons.js", () => {
    
  it("FeelingsRadioButtons renders correctly", () => {
    const button = shallow(<FeelingsRadioButtons />);
    expect(button.length).toBe(1);
  });
    
  it("FeelingsRadioButtons renders correctly across screens", () => {
    const tree = renderer.create(<FeelingsRadioButtons />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing FeelingsRadioButtons.js component interaction", () => {

    it("Press very sad should invoke onPress", () => {
      const mockFn = jest.fn(() => "Pressed very sad");
      const initialProps = {
        setOverallFeeling: () => mockFn(),
      }
      const wrapper = render(<FeelingsRadioButtons {...initialProps} />);
      const verySad = wrapper.getByTestId("VERY_SAD");
      expect(mockFn).not.toHaveBeenCalled();
      fireEvent.press(verySad)
      expect(mockFn).toHaveBeenCalled()
    });
  
    it("Press sad should invoke onPress", () => {
      const mockFn = jest.fn(() => "Pressed sad");
      const initialProps = {
        setOverallFeeling: () => mockFn(),
      }
      const wrapper = render(<FeelingsRadioButtons {...initialProps} />);
      const verySad = wrapper.getByTestId("SAD");
      expect(mockFn).not.toHaveBeenCalled();
      fireEvent.press(verySad)
      expect(mockFn).toHaveBeenCalled()
    });

    it("Press neutral should invoke onPress", () => {
      const mockFn = jest.fn(() => "Pressed neutral");
      const initialProps = {
        setOverallFeeling: () => mockFn(),
      }
      const wrapper = render(<FeelingsRadioButtons {...initialProps} />);
      const verySad = wrapper.getByTestId("NEUTRAL");
      expect(mockFn).not.toHaveBeenCalled();
      fireEvent.press(verySad)
      expect(mockFn).toHaveBeenCalled()
    });

    it("Press happy should invoke onPress", () => {
      const mockFn = jest.fn(() => "Pressed happy");
      const initialProps = {
        setOverallFeeling: () => mockFn(),
      }
      const wrapper = render(<FeelingsRadioButtons {...initialProps} />);
      const verySad = wrapper.getByTestId("HAPPY");
      expect(mockFn).not.toHaveBeenCalled();
      fireEvent.press(verySad)
      expect(mockFn).toHaveBeenCalled()
    });

    it("Press very Happy should invoke onPress", () => {
      const mockFn = jest.fn(() => "Pressed very happy");
      const initialProps = {
        setOverallFeeling: () => mockFn(),
      }
      const wrapper = render(<FeelingsRadioButtons {...initialProps} />);
      const verySad = wrapper.getByTestId("VERY_HAPPY");
      expect(mockFn).not.toHaveBeenCalled();
      fireEvent.press(verySad)
      expect(mockFn).toHaveBeenCalled()
    });
  
});