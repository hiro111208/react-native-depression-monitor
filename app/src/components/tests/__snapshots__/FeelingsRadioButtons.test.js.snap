// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FeelingsRadioButtons.js FeelingsRadioButtons renders correctly across screens 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "#ffa351ff",
      "borderRadius": 50,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "width": "95%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="VERY_SAD"
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/verySad.png",
        }
      }
      style={
        Object {
          "height": 50,
          "marginHorizontal": 7,
          "marginVertical": 7,
          "width": 50,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="SAD"
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/sad.png",
        }
      }
      style={
        Object {
          "height": 50,
          "marginHorizontal": 7,
          "marginVertical": 7,
          "width": 50,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="NEUTRAL"
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/neutral.png",
        }
      }
      style={
        Object {
          "height": 50,
          "marginHorizontal": 7,
          "marginVertical": 7,
          "width": 50,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="HAPPY"
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/happy.png",
        }
      }
      style={
        Object {
          "height": 50,
          "marginHorizontal": 7,
          "marginVertical": 7,
          "width": 50,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="VERY_HAPPY"
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/veryHappy.png",
        }
      }
      style={
        Object {
          "height": 50,
          "marginHorizontal": 7,
          "marginVertical": 7,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;
